### PR TITLE
DocumentEditor: gallery enchancement

### PR DESCRIPTION
**Pull request by Vasya-Zo**

### CODE REVIEW

File: src/form/editors/impl/document/gallery/templates/galleryWindow.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
...
-  <div class="js-image"></div>
-</div>
-<div class="js-download btn btn-pale"><span>{{localize 'CORE.FORM.EDITORS.DOCUMENT.DOWNLOAD'}}</span></div>
-<div class="js-loading-region dev-l-loader-image-gallery"></div>
\ No newline at end of file
+<div class="js-top-menu-region"></div>
+<div class="js-info-widget-region"></div>
+<div class="js-image-region gallery-main-image"></div>
+<div class="js-download-panel"></div>
+<div class="js-previews-collection-region gallery-previews-collection"></span></div>
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/gallery/templates/galleryWindow.html]: iconPrefixer

File: src/form/editors/impl/document/gallery/views/DownloadPanelView.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        'click @ui.deleteButton': '__onDeleteBtnClick',
+        'click @ui.downloadButton': '__onDownloadBtnClick'
+    },
+
+    onRender() {
+        this.getRegion('fileNameRegion').show(new FileNameView({ model: this.model }));
+    },
+
+    __onDownloadBtnClick() {
+        this.reqres.request('image:download', this.model);
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/gallery/views/DownloadPanelView.js]: !

File: resources/styles/gallery.css [hunk 3]
```diff
@@ -1,130 +1,221 @@
...
-.galleryThumbnailInfo {
-    position : absolute;
-    height : 20px;
-    color : white;
-    text-align: center;
-    width : 100%;
-    bottom : 0;
+    height: 100%;
+    margin-bottom: 20px;
+    z-index: -42;
```
Comment (stasguryev) [inline resources/styles/gallery.css]: ?

File: resources/styles/gallery.css [hunk 4]
```diff
@@ -1,130 +1,221 @@
...
+
+.info-widget-text-container {
+    font-size: 10px;
+    color: white;
+    width: 100%;
+    margin-left: 5px;
+}
+
+.js-info-widget-text {
+    font-weight: bold;
```
Comment (stasguryev) [inline resources/styles/gallery.css]: !

File: src/form/editors/impl/document/gallery/views/GalleryWindowView.js [hunk 5]
```diff
@@ -1,91 +1,159 @@
...
+        }
+        this.topMenuView.trigger('fullscreen:mode:change', isFullScreen);
     },
 
     __keyAction(event) {
-        if (event.keyCode === 27) {
-            this.__onClose();
+        event.stopPropagation();
+        switch (event.keyCode) {
+            case 27: //Esc
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/gallery/views/GalleryWindowView.js]: utils/keyCode

File: src/form/editors/impl/document/gallery/templates/galleryWindow.html (hunk 1 above)
Comment (stasguryev) [inline src/form/editors/impl/document/gallery/templates/galleryWindow.html]: remove </span>

File: src/form/editors/impl/document/gallery/views/GalleryWindowView.js [hunk 6]
```diff
@@ -1,91 +1,159 @@
...
 
-    __addImage(model) {
-        this.ui.image.empty().append(this.reqres.request('image:get', model));
+    __toggleFullScreenClass(action) {
+        this.el.classList[action]('fullscreen-mode');
+    },
+
+    __onfullscreenchange() {
+        const isFullScreen = document.fullscreenElement;
+        if (document.fullscreenElement) {
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/gallery/views/GalleryWindowView.js]: toggle first argument - state

File: src/form/editors/impl/document/models/DocumentModel.js [hunk 7]
```diff
@@ -1,7 +1,27 @@
...
             this.set({ uniqueId: id || streamId });
         }
+
+        _.extend(this, new SelectableBehavior.Selectable(this));
+    },
+
+    __getPreviewTag() {
+        const extension = this.get('extension');
+        if (extension && typeof extension === 'string') {
+            if (graphicFileExtensions.indexOf(extension.toLowerCase()) > -1) {
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/models/DocumentModel.js]: graphicFileExtensions.indexOf(extension.toLowerCase()) > -1 === graphicFileExtensions.includes(extension.toLowerCase())

File: src/form/editors/impl/document/models/DocumentModel.js [hunk 8]
```diff
@@ -1,7 +1,27 @@
...
         if (!uniqueId) {
             this.set({ uniqueId: id || streamId });
         }
+
+        _.extend(this, new SelectableBehavior.Selectable(this));
+    },
+
+    __getPreviewTag() {
+        const extension = this.get('extension');
+        if (extension && typeof extension === 'string') {
```
Comment (PavelKoroteev) [inline src/form/editors/impl/document/models/DocumentModel.js]: if (typeof extension === 'string')

File: src/form/editors/impl/document/gallery/views/GalleryWindowView.js [hunk 9]
```diff
@@ -1,91 +1,159 @@
...
-        if (visible) {
-            loadingRegion.show(new LoadingView());
-        } else {
-            loadingRegion.reset();
+    __onPreviousImage() {
+        this.__selectNext('previous');
+    },
+
+    __onNextImage() {
+        this.__selectNext('next');
```
Comment (stasguryev) [inline src/form/editors/impl/document/gallery/views/GalleryWindowView.js]: directions in const

File: src/form/editors/impl/document/gallery/views/GalleryWindowView.js [hunk 10]
```diff
@@ -1,91 +1,159 @@
...
-    __onImageClick() {
-        const collection = this.options.imagesCollection;
-        let index = collection.indexOf(this.model);
-        if (index === collection.length - 1) {
-            index = 0;
+    __toggleZoomMode() {
+        const imageElement = this.getRegion('mainImageRegion').el;
+        const isZoomedIn = imageElement.classList.contains('zoomed-in');
+        if (isZoomedIn) {
+            imageElement.classList.remove('zoomed-in');
```
Comment (stasguryev) [inline src/form/editors/impl/document/gallery/views/GalleryWindowView.js]: const class

File: src/form/editors/impl/document/gallery/views/ImageView.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+        loadingRegion: '.js-gallery-loading-region'
+    },
+
+    className() {
+        const isVideo = this.model.get('previewTag') === 'video' ? playIconClassName : '';
+        return `gallery-image ${isVideo}`;
+    },
+
+    events: {
+        click: '__onClick'
```
Comment (stasguryev) [inline src/form/editors/impl/document/gallery/views/ImageView.js]: triggers: { click: 'item:click' }

File: src/form/editors/impl/document/gallery/views/InfoWidgetView.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+export default WidgetBaseView.extend({
+    className: 'gallery-info-widget',
+
+    template: Handlebars.compile(template),
+
+    templateContext() {
+        const infoWidget = this.model.get('infoWidget');
+
+        if (infoWidget) {
```
Comment (stasguryev) [inline src/form/editors/impl/document/gallery/views/InfoWidgetView.js]: return this.model.get('infoWidget')

File: src/form/editors/impl/document/meta.js [hunk 13]
```diff
@@ -20,15 +65,11 @@ export default {
...
-                    type ||
-                    getExtensionFromName(name);
+        const ext = extension || type || getExtensionFromName(name);
         let icon;
 
         if (ext) {
-            icon = Object.keys(fileIconClasses).find(key =>
-                fileIconClasses[key].includes(ext.toLowerCase())
-            );
+            icon = Object.keys(fileIconClasses).find(key => fileIconClasses[key].includes(ext.toLowerCase()));
```
Comment (stasguryev) [inline src/form/editors/impl/document/meta.js]: includes => ===

File: demo/public/app/cases/editors/DocumentEditor.js [hunk 14]
```diff
@@ -9,22 +9,66 @@ export default function() {
...
             },
             {
-                extension: 'jpg',
+                extension: 'mp4',
                 id: 'document.2',
+                name: 'SampleVideo',
+                url: 'images/SampleVideo.mp4',
+                infoWidget: {
+                    text: '11.11.2111 12:59',
+                    subtext: 'Pavel Koroteev',
```
Comment (PavelKoroteev) [inline demo/public/app/cases/editors/DocumentEditor.js]: Вошел в историю)

**Pull request merged**